src/eduid_webapp/security/views/reset_password.py

```diff
@@ -103,23 +104,26 @@ def reset_password():
         'errors': [],
     }
     if request.method == 'POST':
-        form = ResetPasswordEmailSchema().load(request.form)
-        if not form.errors and session.get_csrf_token() == form.data['csrf']:
-            current_app.logger.info('Trying to send password reset email to {}'.format(form.data['email']))
-            try:
-                send_password_reset_mail(form.data['email'])
-            except MailTaskFailed as e:
-                current_app.logger.error('Sending e-mail failed: {}'.format(e))
-                view_context = {
-                    'heading': _('Temporary technical problem'),
-                    'text': _('Please try again.'),
-                    'retry_url': url_for('reset_password.reset_password'),
-                    'retry_url_txt': _('Try again'),
-                }
-                return render_template('error.jinja2', view_context=view_context)
-            view_context['form_post_success'] = True
-            view_context['text'] = ''
-        view_context['errors'] = form.errors
+        try:
+            form = ResetPasswordEmailSchema().load(request.form)
+            if session.get_csrf_token() == form['csrf']:
```

Comment:
This try/except covers much more code than it ought to. Most of this code should be moved to an else: clause after except: I think.